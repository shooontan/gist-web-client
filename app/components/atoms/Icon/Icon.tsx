import { format } from 'url';
import * as React from 'react';
import Link, { LinkProps } from 'next/link';
import styled from 'styled-components';
import pure from 'recompose/pure';

interface Props extends React.ImgHTMLAttributes<HTMLImageElement> {
  src: string;
  href?: LinkProps['href'];
}

const Icon = (props: Props) => {
  const { src, href, ...rest } = props;
  if (!href) {
    return <Image src={src} {...rest} />;
  }
  return (
    <Link href={href}>
      <Anker href={format(href)}>
        <Image src={src} {...rest} />
      </Anker>
    </Link>
  );
};

export default pure(Icon);

Icon.defaultProps = {
  width: 50,
  height: 50,
};

const Image = styled.img`
  border-radius: 3px;
  vertical-align: bottom;
`;

const Anker = styled.a`
  display: inline-block;
  cursor: pointer;
`;
