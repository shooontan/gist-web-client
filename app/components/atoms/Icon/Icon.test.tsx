import { render } from 'react-testing-library';
import Icon from './Icon';

const src = 'https://example.com/icon.png';
const href = 'https://example.com/';
const hrefUser = {
  pathname: '/user',
  query: { name: 'username' },
};

test('snapshot Icon', () => {
  const { container } = render(<Icon src={src} />);
  expect(container.firstChild).toMatchSnapshot();

  const { container: hrefContainer } = render(<Icon src={src} href={href} />);
  expect(hrefContainer.firstChild).toMatchSnapshot();

  const { container: paramsHrefContainer } = render(
    <Icon src={src} href={hrefUser} />
  );
  expect(paramsHrefContainer.firstChild).toMatchSnapshot();

  const { container: attrComtainer } = render(
    <Icon src={src} width="400" height="400" alt="icon" />
  );
  expect(attrComtainer.firstChild).toMatchSnapshot();
});
