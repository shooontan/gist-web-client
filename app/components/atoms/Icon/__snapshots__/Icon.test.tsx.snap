// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Icon 1`] = `
.c0 {
  border-radius: 3px;
  vertical-align: bottom;
}

<img
  class="c0"
  height="50"
  src="https://example.com/icon.png"
  width="50"
/>
`;

exports[`snapshot Icon 2`] = `
.c1 {
  border-radius: 3px;
  vertical-align: bottom;
}

.c0 {
  display: inline-block;
  cursor: pointer;
}

<a
  class="c0"
  href="https://example.com/"
>
  <img
    class="c1"
    height="50"
    src="https://example.com/icon.png"
    width="50"
  />
</a>
`;

exports[`snapshot Icon 3`] = `
.c1 {
  border-radius: 3px;
  vertical-align: bottom;
}

.c0 {
  display: inline-block;
  cursor: pointer;
}

<a
  class="c0"
  href="/user?name=username"
>
  <img
    class="c1"
    height="50"
    src="https://example.com/icon.png"
    width="50"
  />
</a>
`;

exports[`snapshot Icon 4`] = `
.c0 {
  border-radius: 3px;
  vertical-align: bottom;
}

<img
  alt="icon"
  class="c0"
  height="400"
  src="https://example.com/icon.png"
  width="400"
/>
`;
